In [0]:
# Class IT-533 Week #2  - Introduction to Basic Analysis with R and Python
# Summer 2020
# Week #2 Practice Code
# By Wei Geng(David)
# Created on: 05/25/2020

### ----------------------------------------------Question 1 Code-----------------------------------------------------

In [0]:
# define the manhattan distance user-defined function, and later I can recall the function to return the value.
from math import*
def manhattan_distance(x,y):
    return sum(abs(a-b) for a,b in zip(x,y))

In [3]:
# import needed package and upload the file to google colab
from google.colab import files
uploaded = files.upload()

Saving adult.data.simplified.csv to adult.data.simplified (1).csv


In [4]:
# Convert to pandas DataFrame
import io
import pandas as pd
df = pd.read_csv(io.BytesIO(uploaded['adult.data.simplified.csv']))
print(df.info(verbose=True))
df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32561 entries, 0 to 32560
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   age             32561 non-null  int64 
 1   workclass       32561 non-null  object
 2   education       32561 non-null  object
 3   educationyears  32561 non-null  int64 
 4   maritalstatus   32561 non-null  object
 5   occupation      32561 non-null  object
 6   relationship    32561 non-null  object
 7   race            32561 non-null  object
 8   sex             32561 non-null  object
 9   hoursperweek    32561 non-null  int64 
 10  nativecountry   32561 non-null  object
 11  incomeUSD       32561 non-null  int64 
dtypes: int64(4), object(8)
memory usage: 3.0+ MB
None


age         workclass  education  ...  hoursperweek  nativecountry incomeUSD
0   39         State-gov  Bachelors  ...            40  United-States     43747
1   50  Self-emp-not-inc  Bachelors  ...            13  United-States     38907
2   38           Private    HS-grad  ...            40  United-States     25055
3   53           Private       11th  ...            40  United-States     26733
4   28           Private  Bachelors  ...            40           Cuba     23429

[5 rows x 12 columns]

In [5]:
# Check the value count of the education attributes as well as its distributions
print(df.shape)
print(df.education.value_counts(dropna=False))
df.education.value_counts(dropna=False, normalize=True)

(32561, 12)
HS-grad         10501
Some-college     7291
Bachelors        5355
Masters          1723
Assoc-voc        1382
11th             1175
Assoc-acdm       1067
10th              933
7th-8th           646
Prof-school       576
9th               514
12th              433
Doctorate         413
5th-6th           333
1st-4th           168
Preschool          51
Name: education, dtype: int64


HS-grad         0.322502
Some-college    0.223918
Bachelors       0.164461
Masters         0.052916
Assoc-voc       0.042443
11th            0.036086
Assoc-acdm      0.032769
10th            0.028654
7th-8th         0.019840
Prof-school     0.017690
9th             0.015786
12th            0.013298
Doctorate       0.012684
5th-6th         0.010227
1st-4th         0.005160
Preschool       0.001566
Name: education, dtype: float64

In [6]:
# Convert the nominal variable to numeric categorical variable. Since the education attributes has 17 levels, I mapped it to the numeric value the represent each level. 
# So that it can be calculated for the Manhattan Distance
df['education_num']=df.education.map({'Preschool':1,'1st-4th':2,'5th-6th':3,'7th-8th':4,'9th':5,'10th':6,'11th':7,
                  '12th':9,'HS-grad':10,'Assoc-voc':11,'Assoc-acdm':12,'Some-college':13,'Bachelors':14,
                  'Masters':15,'Doctorate':16,'Prof-school':17})
df['education_num'].value_counts()

10    10501
13     7291
14     5355
15     1723
11     1382
7      1175
12     1067
6       933
4       646
17      576
5       514
9       433
16      413
3       333
2       168
1        51
Name: education_num, dtype: int64

In [7]:
# call the manhattan_distance funtion and set the parameter as the first 10 datapoints of the x column "education", and y column "incomeUSD". Return the distance value. 
print('Manhattan Distance of the first 10 datapoints of x and y are: ',manhattan_distance(df['education_num'][0:10],df['incomeUSD'][0:10]))

Manhattan Distance of the first 10 datapoints of x and y are:  495544


### ----------------------------------------------Question 2 Code-----------------------------------------------------

In [8]:
# Check the value count of the maritalstatus attributes as well as its distributions
print(df.shape)
print(df.maritalstatus.value_counts(dropna=False))
df.maritalstatus.value_counts(dropna=False, normalize=True)

(32561, 13)
Married-civ-spouse       14976
Never-married            10683
Divorced                  4443
Separated                 1025
Widowed                    993
Married-spouse-absent      418
Married-AF-spouse           23
Name: maritalstatus, dtype: int64


Married-civ-spouse       0.459937
Never-married            0.328092
Divorced                 0.136452
Separated                0.031479
Widowed                  0.030497
Married-spouse-absent    0.012837
Married-AF-spouse        0.000706
Name: maritalstatus, dtype: float64

In [9]:
# Convert the nominal variable to numeric categorical variable. Since the maritalstatus attributes has 7 levels, I mapped it to the numeric value the represent each level. 
# So that it can be calculated for the Manhattan Distance
df['maritalstatus_num']=df.maritalstatus.map({'Married-civ-spouse':1,'Never-married':2,'Divorced':3,'Separated':4,'Widowed':5,'Married-spouse-absent':6,'Married-AF-spouse':7,})
df['maritalstatus_num'].value_counts()

1    14976
2    10683
3     4443
4     1025
5      993
6      418
7       23
Name: maritalstatus_num, dtype: int64

In [10]:
# call the manhattan_distance funtion and set the parameter as the first 10 datapoints of the x column "maritalstatus", and y column "incomeUSD". Return the distance value. 
print('Manhattan Distance of the first 10 datapoints of x and y are: ',manhattan_distance(df['maritalstatus_num'][0:10],df['incomeUSD'][0:10]))

Manhattan Distance of the first 10 datapoints of x and y are:  495643


#### By looking at the values of Manhattan Distance from part 1 and 2, we can conclude that Education is more closely related to incomeUSD since it has smaller value, 495544. In reality, education does have more impact and more weight on personal income, and that is why we need to pursue higher education to get more financial return.